pkg/expr/sql/db.go

```diff
@@ -90,40 +94,110 @@ func convertToDataFrame(iter gomysql.RowIter, schema gomysql.Schema, f *data.Fra
 	return nil
 }
 
-func (db *DB) QueryFramesInto(name string, query string, frames []*data.Frame, f *data.Frame) error {
-	// TODO: Consider if this should be moved outside of this function
-	// or indeed into convertToDataFrame
-	f.Name = name
+// TODO: Check if it really makes sense to receive a shared context here, rather than creating a new one
+func (db *DB) writeDataframeToDb(ctx *mysql.Context, tableName string, frame *data.Frame) error {
```

Comment:
`frame` is getting mutilated here, I think it is going through SSE conversion, we need that not to happen generally when it goes into SQL (unless for future cases we set up).

Reply:
Mutilated **before** it lands here, is that what you mean?
When I first read this I thought you were critiquing the code in this function 😢 - I think it because clearer when we spoke yesterday

Reply:
> Mutilated before it lands here, is that what you mean?
Yes, sorry for that.